### ISSUE TITLE
[depth_from_video_in_the_wild] cannot load from ImageNet ckpt

### ISSUE THREAD

**Issue by yanfengliu** [google-research/google-research#130]
It seems that the batchnorm weights were missing from the checkpoint. Loading from the checkpoint gives the error `Not found: Key conv1/bn/beta not found in checkpoint` @gariel-google

**Comment by yanfengliu** [comment]
https://stackoverflow.com/questions/45800871/tensorflow-save-restore-batch-norm

**Comment by yanfengliu** [comment]
Upon further inspection, it appears to be having issues with all weights, not just batchnorm. I couldn't load the kernel weights for the convolution layers either.

**Comment by yanfengliu** [comment]
A second issue is that the network gets stuck on certain examples when training from scratch. The exact input that it gets stuck on is different every run. @gariel-google

**Comment by gariel-google** [comment]
@yanfengliu I am sorry for the delayed response. Regarding loading the
imagene checkpoint, I admit that I am not 100% sure how to import it from
the pytorch checkpoint, but perhaps this code
https://github.com/tensorflow/models/blob/master/research/struct2depth/util.py#L215
can
provide a clue? These seems to be some special treatment there for the
batchnorm variables. Unfortunately I do not have access to the code that
was used to import the imagenet checkpoint, and I m not a pytorch user
myself, so I am sorry that I cannot help more.

Regarding the training getting stuck - could you explain more? Does it
crash? Or just hangs? We never observed anything like that.

**Comment by yanfengliu** [comment]
@gariel-google Thank you for your reply! The program doesn't crash. It just hangs. I tried to train with struct2depth and it had the same problem. Looks like this might have been inherited from it. I will keep investigating.

**Comment by pigletsc** [comment]
@yanfengliu @gariel-google
Hi,
I got same issue, when training from scratch, there is Error "LossTensor is inf or nan: Tensor had NaN values" (it happend sometimes)

And I can not load the imageNet pretrained model like @yanfengliu siad, I download the imagenet pretrain model here: [Issue](https://github.com/facebookresearch/agem/issues/4)

**Comment by suvigy** [comment]
@yanfengliu @pigletsc 
Assuming you converted the torch model to tensorflow model and are having the following files for imagenet checkpoint:

init/
  checkpoint
  model.ckpt.data-00000-of-00001 
  model.ckpt.index
  model.ckpt.meta

in this case you have to specify --imagenet_ckpt=.../init/model.ckpt 
So not even an existing filename (and not the .index file, or the init dir)
That works for me

@pigletsc 
Regarding 'Tensor had NaN values'. Issue #162  is about this problem. There is an important bug in CUDA 10.0, and the related tensorflow 1.x (x>=13.1) binaries are built with CUDA 10.0 til. We built tensorflow 14.0 with CUDA 10.1. It works that way. Or you can try the training only on CPU. :)

**Comment by gariel-google** [comment]
suvigy@ Thanks so much for your help!

**Comment by poornimajd** [comment]
Great work @gariel-google and team! 
As mentioned below,I need to convert the torch model to tensorflow.



> Assuming you converted the torch model to tensorflow model and are having the following files for imagenet checkpoint:
> 
> init/
> checkpoint
> model.ckpt.data-00000-of-00001
> model.ckpt.index
> model.ckpt.meta
> 
> in this case you have to specify --imagenet_ckpt=.../init/model.ckpt
> So not even an existing filename (and not the .index file, or the init dir)
> That works for me

In regard to this , I am not sure as to where exactly is the torch model which is to be used as imagenet pretrained weights(after conversion.)
Any suggestion is greatly appreciated!
Thank you

**Comment by gariel-google** [comment]
@poornimajd AFAIK, the imagenet imagenet checkpoint is from [here] (https://pytorch.org/docs/stable/torchvision/models.html) (resnet18), but honestly I'm not 100% sure. 

Sure, you can initialize from a cityscapes model. The existing code doesn't expose this option as a flag, but an easy hack would be top copy the cityscapes checkpoint into the training directory and then start the training.

**Comment by adizhol** [comment]
@poornimajd I tried to use ONNX, but that didn't work.
Maybe this will work for you: https://github.com/dalgu90/resnet-18-tensorflow

**Comment by surabhi96** [comment]
@adizhol did you finally manage to convert a torch model to a tensorflow model?

**Comment by adizhol** [comment]
Not really. I managed to load them, but the output is different
